src/libraries/Microsoft.Extensions.Logging.Abstractions/tests/Microsoft.Extensions.Logging.Generators.Tests/LoggerMessageGeneratorParserTests.cs

```diff
@@ -616,6 +616,19 @@ partial class C
             Assert.Equal(DiagnosticDescriptors.LoggingMethodIsGeneric.Id, diagnostics[0].Id);
         }
 
+        [Fact]
+        public async Task SupportsRefKindIn()
+        {
+            IReadOnlyList<Diagnostic> diagnostics = await RunGenerator(@"
+                partial class C
+                {
+                    [LoggerMessage(EventId = 0, Level = LogLevel.Debug, Message = ""Parameter {P1}"")]
+                    static partial void M(ILogger logger, in int p1);
```

Comment:
Do we already handle / test `ref int` as well?

Reply:
I was just about to write that for `ref` and `out` I would make a PR separately, that would need to have a diagnostic alongside it because for ref or out we would just not want to keep the method

Reply:
so basically if the log method has ref or out we would not support it and just give out the diagnostic

Reply:
I understand why it would make sense to just fail for `out`.  Why doesn't it make sense to support `ref`?  The implementation wouldn't be modifying it, but `ref` is often used in such situations... and `in` is just a `ref` under the covers with some additional metadata sprinkled on top.

Reply:
sure thanks, updated PR.